core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/downscale/ClusterDownscaleActions.java

```diff
@@ -69,15 +77,23 @@ protected void doExecute(ClusterViewContext context, ClusterDownscaleTriggerEven
             @Override
             protected void doExecute(ClusterViewContext context, CollectDownscaleCandidatesResult payload, Map<Object, Object> variables) {
                 variables.put(ContextKeys.PRIVATE_IDS, payload.getPrivateIds());
-                Boolean repair = (Boolean) variables.get(REPAIR);
-                Stack stack = stackService.getByIdWithListsInTransaction(context.getStackId());
                 Selectable event;
-                DecommissionRequest decommissionRequest =
-                        new DecommissionRequest(context.getStackId(), payload.getHostGroupName(), payload.getPrivateIds(), payload.getRequest().getDetails());
-                if (repair && stack.hasCustomHostname()) {
-                    event = new DecommissionResult(decommissionRequest, getHostNamesForPrivateIds(payload.getPrivateIds(), stack));
+                if (isPayloadContainsPrivateAnyId(payload)) {
```

Comment:
Did you mean isPayloadContainsAnyPrivateId?

Reply:
yeah 😅